rules.js

```diff
@@ -186,6 +186,22 @@ class sub_with_return extends rule(3) {
     }
 }
 
+class yoda_condition extends rule(3) {
+    constructor() {
+        super()
+        this._node = 'bop'
+        this.message = `in conditional expression constant should be on the right`
+    }
+
+    check(node) {
+        const compareOp = /^[>=<]{1,2}$/.test(node.op)
```

Comment:
not sure it matters, but this will match `<<` or `>>`

Reply:
Hmm. Good point about shift operator. Having number of the left side of shift is actually valid. For example: `1 << bits`

Reply:
I would think that shift operations shouldn't be allowed in conditionals right? Similarly, this would match on `==` but looks like that's not a valid comparison. Although like Kev says, maybe doesn't really matter since the purpose of this check is just for the ordering.

Reply:
Shift operator can show up in conditional expression: `if 1 << bits > 128 then ...`

Reply:
https://developer.roku.com/docs/references/brightscript/language/expressions-variables-types.md#operators